File: 104-maximum-depth-of-binary-tree/maximum-depth-of-binary-tree.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def maxDepth(self, root: Optional[TreeNode]) -> int:
        if not root:
            return 0
        stack = [(root,1)]
        ret = 0
        while stack:
            curr,level = stack.pop(0)
            if curr.left:
                stack.append((curr.left,level+1))
            if curr.right:
                stack.append((curr.right,level+1))
            ret = max(ret,level)
        return ret

File: 104-maximum-depth-of-binary-tree/maximum-depth-of-binary-tree.ts
/**
 * Definition for a binary tree node.
 * class TreeNode {
 *     val: number
 *     left: TreeNode | null
 *     right: TreeNode | null
 *     constructor(val?: number, left?: TreeNode | null, right?: TreeNode | null) {
 *         this.val = (val===undefined ? 0 : val)
 *         this.left = (left===undefined ? null : left)
 *         this.right = (right===undefined ? null : right)
 *     }
 * }
 */
function dfs(curr:TreeNode|null, count:number){
    if (curr === null){
        return count
        }
    return Math.max(dfs(curr.left,count+1),dfs(curr.right,count+1))
}
function maxDepth(root: TreeNode | null): number {
    return dfs(root,0)
};
